elm_kernel/kernel.py

```diff
@@ -28,22 +32,44 @@ def do_execute(self, code, silent,
 
             #  TODO: Deal with exceptions from this call
             subprocess.run(['elm-make', infile, '--yes',
-                            '--output={}'.format(outfile)],
-                           cwd=tmpdirname,
-                           check=True)
+                        '--output={}'.format(outfile)],
+                       cwd=tmpdirname,
+                       check=True)
+
 
             with open(outfile, mode='rt') as f:
                 javascript = f.read()
 
-        javascript = javascript + "; var mountNode = document.getElementById('elm-div'); Elm.Main.embed(mountNode);"
+
+        div_id = 'elm-div-' + str(self.execution_count)
+
+        javascript = """ 
+            var defineElm = function(cb) {
+                if (this.Elm) { 
+                    this.oldElm = this.Elm;
+                }
+                var define = null;
+            
+        """ + javascript + """
```

Comment:
using triple quotes like this so it looks like JS. Not using format strings since it's easy to catch yourself out when inline JS like this. Done it a few times!